#### About
###### 1
###### 2
###### 3


##### Loading Data
###### The first step is to loading data.

In [2]:
%pip install kagglehub

Note: you may need to restart the kernel to use updated packages.


In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import kagglehub
import os
from IPython.display import display

# easy read 
pd.set_option('display.float_format', lambda x: '%.2f' % x)
# Download latest version
path = kagglehub.dataset_download("artermiloff/steam-games-dataset")
file_path = os.path.join(path, "games_march2025_full.csv")

In [4]:
dataset = pd.read_csv(file_path)

# Display the dataset
display(dataset.head())

# Display basic information about the dataset
display(dataset.info())

,appid,name,release_date,required_age,price,dlc_count,detailed_description,about_the_game,short_description,reviews,...,average_playtime_2weeks,median_playtime_forever,median_playtime_2weeks,discount,peak_ccu,tags,pct_pos_total,num_reviews_total,pct_pos_recent,num_reviews_recent
0,730,Counter-Strike 2,2012-08-21,0,0.00,1,"For over two decades, Counter-Strike has offer...","For over two decades, Counter-Strike has offer...","For over two decades, Counter-Strike has offer...",NaN,...,879,5174,350,0,1212356,"{'FPS': 90857, 'Shooter': 65397, 'Multiplayer'...",86,8632939,82,96473
1,578080,PUBG: BATTLEGROUNDS,2017-12-21,0,0.00,0,"LAND, LOOT, SURVIVE! Play PUBG: BATTLEGROUNDS ...","LAND, LOOT, SURVIVE! Play PUBG: BATTLEGROUNDS ...",Play PUBG: BATTLEGROUNDS for free. Land on str...,NaN,...,0,0,0,0,616738,"{'Survival': 14838, 'Shooter': 12727, 'Battle ...",59,2513842,68,16720
2,570,Dota 2,2013-07-09,0,0.00,2,"The most-played game on Steam. Every day, mill...","The most-played game on Steam. Every day, mill...","Every day, millions of players worldwide enter...",“A modern multiplayer masterpiece.” 9.5/10 – D...,...,1536,898,892,0,555977,"{'Free to Play': 59933, 'MOBA': 20158, 'Multip...",81,2452595,80,29366
3,271590,Grand Theft Auto V Legacy,2015-04-13,17,0.00,0,"When a young street hustler, a retired bank ro...","When a young street hustler, a retired bank ro...",Grand Theft Auto V for PC offers players the o...,NaN,...,771,7101,74,0,117698,"{'Open World': 32644, 'Action': 23539, 'Multip...",87,1803832,92,17517
4,488824,Tom Clancy's Rainbow Six® Siege,2015-12-01,17,19.99,9,Edition Comparison Ultimate Edition The Tom Cl...,“One of the best first-person shooters ever ma...,"Tom Clancy's Rainbow Six® Siege is an elite, t...",NaN,...,0,0,0,0,0,"{'FPS': 8082, 'Multiplayer': 6139, 'Tactical':...",84,1168404,76,13017


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 94948 entries, 0 to 94947
Data columns (total 47 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   appid                     94948 non-null  int64  
 1   name                      94946 non-null  object 
 2   release_date              94948 non-null  object 
 3   required_age              94948 non-null  int64  
 4   price                     94948 non-null  float64
 5   dlc_count                 94948 non-null  int64  
 6   detailed_description      89522 non-null  object 
 7   about_the_game            89499 non-null  object 
 8   short_description         89599 non-null  object 
 9   reviews                   10428 non-null  object 
 10  header_image              94948 non-null  object 
 11  website                   41194 non-null  object 
 12  support_url               44185 non-null  object 
 13  support_email             78848 non-null  object 
 14  window

None

##### Convert to optimally data type for each columns

In [5]:
dataset['release_date'] = pd.to_datetime(dataset['release_date'])

# check
display(dataset['release_date'].head())

0   2012-08-21
1   2017-12-21
2   2013-07-09
3   2015-04-13
4   2015-12-01
Name: release_date, dtype: datetime64[ns]

##### Check for missing values

In [6]:
missing_data = dataset.isnull().sum()
missing_percent = (dataset.isnull().sum() / len(dataset)) * 100

missing_df = pd.DataFrame({
    'Missing Values': missing_data,
    'Percent (%)': missing_percent
})


missing_df = missing_df[missing_df['Missing Values'] > 0].sort_values(by='Missing Values', ascending=False)
missing_df = missing_df.reset_index().rename(columns={'index': 'Column Name'})

missing_df

,Column Name,Missing Values,Percent (%)
0,score_rank,94909,99.96
1,metacritic_url,91372,96.23
2,reviews,84520,89.02
3,notes,78296,82.46
4,website,53754,56.61
5,support_url,50763,53.46
6,support_email,16100,16.96
7,about_the_game,5449,5.74
8,detailed_description,5426,5.71
9,short_description,5349,5.63


##### Check for duplicate values

In [11]:
duplicate_data = dataset.duplicated().sum()
duplicate_percent = (dataset.duplicated().sum() / len(dataset)) * 100

print(f"Duplicates rows: {duplicate_data}")
print(f"Duplizates rows as %: {duplicate_percent}")


# Deep check
duplicate_appids = dataset.duplicated(subset=['appid']).sum()
duplicate_names = dataset.duplicated(subset='name').sum()

print(f"Duplicates 'appid': {duplicate_appids}")
print(f"Duplicates 'name': {duplicate_names}")

Duplicates rows: 0
Duplizates rows as %: 0.0
Duplicates 'appid': 0
Duplicates 'name': 756


##### Found that there 756 duplicates name its may be Remaster games or demo / beta games.
##### I decided to delete demo / beta games.

In [13]:
to_remove = 'demo|beta'

#  orginal dataset is - dataset
dataset_filtered = dataset[~dataset['name'].str.contains(to_remove, case=False,na=False)]

print("before", len(dataset))
print("after", len(dataset_filtered))

before 94948
after 94478


##### Data Description

In [ ]:
dataset.describe()